RealEstate/apps/core/views.py

```diff
@@ -219,3 +222,138 @@ def _permission_check(self, request, role, *args, **kwargs):
+            response_data = {
+                'summary': category.summary,
+                'description': category.description
+            }
+            return HttpResponse(json.dumps(response_data),
+                                content_type="application/json")
+
+        # Renders standard category page
+        homebuyer = request.user.role_object
+        couple = homebuyer.couple
+        categories = Category.objects.filter(couple=couple)
+        weights = CategoryWeight.objects.filter(homebuyer__user=request.user)
+
+        weighted = []
+        for category in categories:
+            missing = True
+            for weight in weights:
+                if weight.category.id is category.id:
+                    weighted.append((category, weight.weight))
+                    missing = False
+                    break
+            if missing:
+                weighted.append((category, None))
+
+        context = {
+            'weights': weighted,
+            'form': AddCategoryForm(),
+            'editForm': EditCategoryForm()
+        }
+        context.update(self._weight_context())
+        return render(request, self.template_name, context)
+
+    def post(self, request, *args, **kwargs):
+        """
+        Depending on what functionality we want, the post may be more of a
+        redirect back to the home page. In that case, much of this code will
+        leave. In the meantime, it saves new data, recreates the same form and
+        posts a success message.
+        """
+        # ajax calls implement weight and delete category commands.
+        if request.is_ajax():
+            id = request.POST['category']
+            category = Category.objects.get(id=id)
+            
+            # Weight a category
+            if request.POST['type'] == 'update':
+                homebuyer = request.user.role_object
+                weight = request.POST['weight']
+                grade, created = CategoryWeight.objects.update_or_create(
+                    homebuyer=homebuyer, category=category,
+                    defaults={'weight': int(weight)})
+                return HttpResponse(json.dumps({"id" : id}),
+                                    content_type="application/json")
+                                    
+            # Delete a category
+            elif request.POST['type'] == 'delete':
+                category.delete()
+                return HttpResponse(json.dumps({"id" : id}),
+                                    content_type="application/json")
+        
+        # Creates or updates a category
+        else:
+            homebuyer = request.user.role_object
+            couple = homebuyer.couple
+            summary = request.POST["summary"]
+            description = request.POST["description"]
+            
+            # Updates a category
+            if "catID" in request.POST:
+                category = get_object_or_404(Category.objects.filter(id=request.POST["catID"]))
```

Comment:
This is part of an if/else block. This if gives "PUT" functionality. The else provides "POST" functionality.